src/library/scala/sys/process/ProcessBuilder.scala

```diff
@@ -166,80 +166,80 @@ trait ProcessBuilder extends Source with Sink {
+  def lineStream(log: ProcessLogger, capacity: Integer): LazyList[String]
 
   /** Starts the process represented by this builder.  The output is returned as
-    * a Stream that blocks when lines are not available but the process has not
+    * a LazyList that blocks when lines are not available but the process has not
     * completed.  Standard error is sent to the console. If the process exits
-    * with a non-zero value, the Stream will provide all lines up to termination
+    * with a non-zero value, the LazyList will provide all lines up to termination
     * but will not throw an exception.
     */
-  def lineStream_! : Stream[String]
+  def lineStream_! : LazyList[String]
 
   /** Starts the process represented by this builder.  The output is returned as
-   * a Stream that blocks when lines are not available but the process has not
+   * a LazyList that blocks when lines are not available but the process has not
    * completed.
    * The producer process will block if the given capacity of lines if filled
    * without being consumed from the stream.
    * Standard error is sent to the console. If the process exits
-   * with a non-zero value, the Stream will provide all lines up to termination
+   * with a non-zero value, the LazyList will provide all lines up to termination
    * but will not throw an exception.
    */
-  def lineStream_!(capacity: Integer): Stream[String]
+  def lineStream_!(capacity: Integer): LazyList[String]
 
   /** Starts the process represented by this builder.  The output is returned as
-    * a Stream that blocks when lines are not available but the process has not
+    * a LazyList that blocks when lines are not available but the process has not
     * completed.  Standard error is sent to the provided ProcessLogger. If the
-    * process exits with a non-zero value, the Stream will provide all lines up
+    * process exits with a non-zero value, the LazyList will provide all lines up
     * to termination but will not throw an exception.
     */
-  def lineStream_!(log: ProcessLogger): Stream[String]
+  def lineStream_!(log: ProcessLogger): LazyList[String]
 
   /** Starts the process represented by this builder.  The output is returned as
-   * a Stream that blocks when lines are not available but the process has not
+   * a LazyList that blocks when lines are not available but the process has not
    * completed.
    * The producer process will block if the given capacity of lines if filled
    * without being consumed from the stream.
    * Standard error is sent to the provided ProcessLogger. If the
-   * process exits with a non-zero value, the Stream will provide all lines up
+   * process exits with a non-zero value, the LazyList will provide all lines up
    * to termination but will not throw an exception.
    */
-  def lineStream_!(log: ProcessLogger, capacity: Integer): Stream[String]
+  def lineStream_!(log: ProcessLogger, capacity: Integer): LazyList[String]
```

Comment:
If we want to have a better backward compatibility we could keep returning a `Stream` but deprecate this operation and introduce a new family of operations prefixed with `lazy` (instead of being suffixed by `Stream`):

~~~ scala
def lazyLines: LazyList[String]
def lazyLines(capacity: Int): LazyList[String]
def lazyLines_!(log: ProcessLoger, capacity: Integer): LazyList[String]
// ...
~~~

What do you think?

Reply:
yes, I like that.

it would be nice to just call the new one `lines`. the reason it isn't called that already is because `sys.process` adds extension methods on `String` so you can write e.g. `import sys.process._; "cat food".lineStream` but `String` already has an extension method called `lines` that just line-splits the string.

I'm half tempted to suggest a design where we avoid bringing collections into it at all and just return an `Iterator` and let people deal with it from there themselves, but idk.

another piece of the puzzle is that `sys.process` is going to become a community-maintained module soon anyway.

Reply:
If we were to make it return an iterator (which I also think is a good idea), what should the methods be called?

Side note: I think allowing you to start a new process with `"command string".lineStream` is unintuitive, bad, and easy to make sensible (e.g. `Process("command string").lineStream`)

Reply:
I think `LazyList` is more convenient than `Iterator` since it supports several traversal.

That being said, an `Iterator` can always be converted to a `LazyList`.

Reply:
Moving to `Iterator` is a significant refactoring and maybe out of scope for this PR (?). One advantage of using an iterator is that consumed lines could be dereferenced.

To me, using `lazyLines` as in your last commit looks good, and it leaves the door open to add `lines: Iterator` versions later.

> Side note: I think allowing you to start a new process with `"command string".lineStream` is unintuitive, bad, and easy to make sensible

👍

Reply:
@lrytz do you want me to make a separate PR deprecating and improving the `"command string".lineStream` situation, as well as potentially working on an `Iterable` implementation? (the latter may actually have to wait a bit due to deprecation windows)

Reply:
> do you want me to

I'm not your boss! 😄

I think (a) separate PR(s) makes sense.